README.adoc

```diff
@@ -171,6 +175,29 @@ nix = {
 NOTE: If you are a https://nixos.org/nix/manual/#ssec-multi-user[trusted user] you may add the
 `nix.conf` lines to `~/.config/nix/nix.conf` instead.
 
+
+==== Cache setup via `.envrc`
+Alternatively you may want to configure the cache only for the plutus project. If you use
+lorri and/or direnv you can do so by adding the following to the `.envrc` at the project root:
+
+----
+# Create a local nix config file
+NIX_CONF_DIR=$(pwd)/.config/nix
+mkdir -p $NIX_CONF_DIR
+
+# Store the iohk specific cache config
+cat - <<EOF > $NIX_CONF_DIR/nix.conf
+substituters        = https://hydra.iohk.io https://iohk.cachix.org https://cache.nixos.org/
+trusted-public-keys = hydra.iohk.io:f/Ea+s+dFdN+3Y/G+FDgSq+a5NEWhJGzdjvKNGv0/EQ= iohk.cachix.org-1:DpRUyj7h7V830dp/i6Nti+NEO2/nhblbov/8MW7Rqoo= cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
+EOF
+
+# Add our configuration to `XDG_CONFIG_DIRS` so that Nix will
+# pick it up and set `NIX_CONF_DIR` to have Nix treat it as
+# configuration file.
```

Comment:
[The manual](https://nixos.org/manual/nix/stable/#description-42) suggests that `NIX_CONF_DIR` controls the system config file, and `XDG_CONFIG_DIRS` controls the user config file (or files?). Also there's a bit more detail in the [master version](https://github.com/NixOS/nix/blob/master/doc/manual/src/command-ref/conf-file-prefix.md).

We probably don't want to stop the system config being used (indeed, that might cause weirdness with the daemon), and ideally we wouldn't stop the user config file being used, we'd just get loaded after it.

So with all of that, I would *guess* that what we want to do is put ourselves at one end of `XDG_CONFIG_DIRS`, but I'm not sure which :D

Reply:
I used this solution for a different client project, took some time to get it to work and as you point out it's different yet again in `master` by now.

The problem is that nix only cares about the **first** config it finds in `XDG_CONFIG_DIRS` so if we add to it to the end it won't work. 

The only other thing that comes to my mind is to take `$(nix show-config)` and add the cache to that..
```
config=$(nix show-config | sed -e s"/.....")
```

would be nicer if the config was in json/yaml/toml/*whatever* ..

Reply:
Ugh. The thing is, I don't *actually* want to clobber the user config. I have useful things in there!

I guess this is fine so long as:
- It's clobbering the user config (I think clobbering the system config might actually break things)
- We warn people that it'll clobber their user config, and not to use this if they care aboutt that

Reply:
I don’t know about “clobber”.. ideally it should do _nothing_ else than adding the iohk cache, something that is a) essential and b) you have to do anyway right?

Admittedly in the current state in this PR it does more than that.

Reply:
Well, if we put a new config file in `XDG_CONFIG_DIRS` and Nix picks that up... presumably that means it *won't* pick up an existing user config file? So any configuration the user has put there will stop applying.

Reply:
Yep, you are right - hence i was wondering about using something awful like `nix show-config | add-cache-config` to add the cache config but *retain all other settings*.

I think i will close this for now and re-open in case i can come up with something that is
- reliable
- only adds cache config and keeps all other settings unchanged

Reply:
I don't mind it so long as it fits the conditions I said above: many people probably don't have user config and so won't care.